packages/opentelemetry-plugin-mongodb/src/mongodb.ts

```diff
@@ -0,0 +1,255 @@
+            ns,
+            command,
+            plugin.patchEnd(span, options as Func<unknown>)
+          );
+        } else {
+          return original.call(
+            this,
+            ns,
+            command,
+            options,
+            plugin.patchEnd(span, callback)
+          );
+        }
+      };
+    };
+  }
+
+  /** Creates spans for Query operations */
+  private getPatchQuery() {
+    const plugin = this;
+    return (original: Func<mongodb.Server>) => {
+      return function(
+        this: mongodb.Server,
+        ns: string,
+        command: MongoInternalCommand,
+        options: {},
+        callback: Func<unknown>
+      ): mongodb.Server {
+        const currentSpan = plugin._tracer.getCurrentSpan();
+        if (currentSpan === null) {
+          return original.apply(this, (arguments as unknown) as unknown[]);
+        }
+        const resultHandler =
+          typeof options === 'function' ? options : callback;
+        if (typeof resultHandler !== 'function') {
+          return original.apply(this, (arguments as unknown) as unknown[]);
+        }
+        const span = plugin._tracer.startSpan(`${ns}.query`, {
+          kind: SpanKind.CLIENT,
+          parent: currentSpan,
+        });
+        if (typeof options === 'function') {
+          return original.call(
+            this,
+            ns,
+            command,
+            plugin.patchEnd(span, options as Func<unknown>)
+          );
+        } else {
+          return original.call(
+            this,
+            ns,
+            command,
+            options,
+            plugin.patchEnd(span, callback)
+          );
+        }
+      };
+    };
+  }
+
+  /** Creates spans for Cursor operations */
+  private getPatchCursor() {
+    const plugin = this;
+    return (original: Func<mongodb.Cursor>) => {
+      return function(
+        this: { ns: string },
+        ...args: unknown[]
+      ): mongodb.Cursor {
+        const currentSpan = plugin._tracer.getCurrentSpan();
+        if (currentSpan === null) {
+          return original.apply(this, (arguments as unknown) as unknown[]);
+        }
+        const resultHandler = args[0] as Func<unknown> | undefined;
+        if (resultHandler === undefined) {
+          return original.apply(this, (arguments as unknown) as unknown[]);
+        }
+        const span = plugin._tracer.startSpan(`${this.ns}.cursor`, {
+          parent: currentSpan,
+          kind: SpanKind.CLIENT,
+        });
+        return original.call(this, plugin.patchEnd(span, resultHandler));
+      };
+    };
+  }
+
+  /**
+   * Ends a created span.
+   * @param span The created span to end.
+   * @param resultHandler A callback function.
+   */
+  patchEnd(span: Span, resultHandler: Func<unknown>): Function {
+    return function patchedEnd(this: {}, ...args: unknown[]) {
+      const error = args[0];
+      if (error instanceof Error) {
+        span.setStatus({
+          code: CanonicalCode.UNKNOWN,
```

Comment:
Why `CanonicalCode.UNKNOWN`?

Reply:
Because it hard to set the correct code depending on the mongodb error, and simpler to just set it as unknown. I don't think it would bring a lot of value to parse the error from mongodb to set the correct error code.